### ISSUE TITLE
Convert exiv2.image.Image to byte or PIL

### ISSUE THREAD

**Issue by umcanpolat** [jim-easterbrook/python-exiv2#4]
First of all, it is a great package, thanks for your effort.

My problem is, I am not using local image, I get image as a byte type, open with exiv2.ImageFactory.open(), use readMetadata() change iptcData() to add/edit/delete value and writeMetadata(). Everything works awesome. After that, I cannot change object type to PIL.Image or bytes with io.BytesIO for upload it back. Maybe I cannot find the way. Thank you.

**Comment by jim-easterbrook** [comment]
I guess the C++ way to do this is with `Image::io()`, but the `BasicIo` class doesn't map easily to Python. You could possibly use `ImageFactory.open` to read a Python `io.BytesIO` object but I don't think it would be able to write back to it.

Yours is a use case I hadn't thought about. (In my use of Exiv2 I sometimes read from buffers but only ever write to files.) I might be able to use `Image::io()::mmap()` to give efficient access to the data as a Python read-only buffer.

**Comment by umcanpolat** [comment]
Thank you for fast respond. When I try to read it with BytesIO, return "no read operation". If there is a common object types between Python and C++, a function convert to this type will be solution. I am waiting news.

**Comment by jim-easterbrook** [comment]
My experiments with `Image::io()::mmap()` are not very promising. It works if `ImageFactory.open` was passed a bytes object, but if a file is opened directly then `mmap` causes a segmentation fault.

**Comment by jim-easterbrook** [comment]
Commit 24e9652 adds Exiv2::BasicIo to the Python interface. I need to do some more testing before issuing a new release.

**Comment by jim-easterbrook** [comment]
Commit 8eaa219 adds memory efficient access to a `BasicIo` instance's data using `mmap()` safely. This allows something like:
```
# get data from somewhere
image = exiv2.ImageFactory.open(data)
# modify metadata
image.writeMetadata()
data = bytes(image.io())
# send data to somewhere
```

**Comment by umcanpolat** [comment]
I tried and it is working. Thanks for fast respond and update.

**Comment by jim-easterbrook** [comment]
In the next release (v0.13.0) the `BasicIo` object returned by `exiv2.Image.io()` has a working `mmap()` method that returns a Python `memoryview` (https://docs.python.org/3/library/stdtypes.html#memoryview) object. The buffer interface to `BasicIo` will be deprecated and eventually withdrawn.

**Comment by jim-easterbrook** [comment]
I've just released v0.13.0. `mmap()` now returns an object that has the buffer interface. This can be used directly by many things that expect `bytes`, or a `memoryview` can be used to give copy-free access to the data.
